cmd/meroxa/builder/builder.go

```diff
@@ -586,3 +596,43 @@ func buildCommandWithSubCommands(cmd *cobra.Command, c Command) {
 		cmd.AddCommand(sub)
 	}
 }
+
+func hasFeatureFlag(flags []string, f string) bool {
+	for _, v := range flags {
+		if v == f {
+			return true
+		}
+	}
+
+	return false
+}
+
+func buildCommandWithFeatureFlag(cmd *cobra.Command, c Command) {
+	v, ok := c.(CommandWithFeatureFlag)
+	if !ok {
+		return
+	}
+
+	// Considering a command with feature flags not ready to be documented and
+	// publicly available.
+	cmd.Hidden = true
+
+	old := cmd.PreRunE
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		if old != nil {
+			err := old(cmd, args)
+			if err != nil {
+				return err
+			}
+		}
+
+		flagRequired := v.FeatureFlag()
+		userFeatureFlags := global.Config.GetStringSlice(global.UserFeatureFlagsEnv)
+
+		if !hasFeatureFlag(userFeatureFlags, flagRequired) {
+			return fmt.Errorf("your account does not have access to the %q feature. Reach out to support@meroxa.com for more information", flagRequired)
+		}
```

Comment:
Can this be moved outside of the `PreRunE` function? I think we already have all the info at that point, don't we? If we do, then we could already detect if the feature flag is enabled and un-hide the command (`cmd.Hidden = false`).

Reply:
Oh no, I see we read the config only after all commands are built ([here](https://github.com/meroxa/cli/blob/ac1587b0a1ff4f11e85bad081340eeeeb15287cf/cmd/meroxa/global/global.go#L58)). We could consider reading the config before even building the commands ([here](https://github.com/meroxa/cli/blob/ac1587b0a1ff4f11e85bad081340eeeeb15287cf/cmd/meroxa/root/root.go#L46)).

Reply:
For context, @lovromazgon and I paired today to explore this option and we both agreed that the effort to make it work is currently not worth it. Maybe in a future iteration we'll revisit this.